pkg/monitoring/migrationstats/collector.go

```diff
@@ -0,0 +1,151 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package migrationstats
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/tools/cache"
+
+	k6tv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+const (
+	PendingMigrations    = "kubevirt_migrate_vmi_pending_count"
+	SchedulingMigrations = "kubevirt_migrate_vmi_scheduling_count"
+	RunningMigrations    = "kubevirt_migrate_vmi_running_count"
+	SucceededMigrations  = "kubevirt_migrate_vmi_succeeded_total"
+	FailedMigrations     = "kubevirt_migrate_vmi_failed_total"
+)
+
+var (
+	migrationMetrics = map[string]*prometheus.Desc{
+		PendingMigrations: prometheus.NewDesc(
+			PendingMigrations,
+			"Number of current pending migrations.",
+			nil,
+			nil,
+		),
+		SchedulingMigrations: prometheus.NewDesc(
+			SchedulingMigrations,
+			"Number of current scheduling migrations.",
+			nil,
+			nil,
+		),
+		RunningMigrations: prometheus.NewDesc(
+			RunningMigrations,
+			"Number of current running migrations.",
+			nil,
+			nil,
+		),
+		SucceededMigrations: prometheus.NewDesc(
+			SucceededMigrations,
+			"Number of migrations successfully executed.",
+			[]string{"vmi", "vmim"},
+			nil,
+		),
+		FailedMigrations: prometheus.NewDesc(
+			FailedMigrations,
+			"Number of failed migrations.",
+			[]string{"vmi", "vmim"},
+			nil,
+		),
+	}
+)
+
+type MigrationCollector struct {
+	migrationInformer cache.SharedIndexInformer
+	clusterConfig     *virtconfig.ClusterConfig
```

Comment:
I did not see any usage of the `clusterConfig`. Is it a leftover? Thanks

Reply:
removed